Mathlib/Data/Stream/Init.lean

```diff
@@ -429,19 +438,42 @@ theorem mem_of_mem_even (a : α) (s : Stream' α) : a ∈ even s → a ∈ s :=
 theorem mem_of_mem_odd (a : α) (s : Stream' α) : a ∈ odd s → a ∈ s := fun ⟨n, h⟩ =>
   Exists.intro (2 * n + 1) (by rw [h, get_odd])
 
-theorem nil_append_stream (s : Stream' α) : appendStream' [] s = s :=
+@[simp] theorem nil_append_stream (s : Stream' α) : appendStream' [] s = s :=
   rfl
 
 theorem cons_append_stream (a : α) (l : List α) (s : Stream' α) :
     appendStream' (a::l) s = a::appendStream' l s :=
   rfl
 
-theorem append_append_stream : ∀ (l₁ l₂ : List α) (s : Stream' α),
+@[simp] theorem append_append_stream : ∀ (l₁ l₂ : List α) (s : Stream' α),
     l₁ ++ l₂ ++ₛ s = l₁ ++ₛ (l₂ ++ₛ s)
   | [], _, _ => rfl
   | List.cons a l₁, l₂, s => by
     rw [List.cons_append, cons_append_stream, cons_append_stream, append_append_stream l₁]
 
+lemma get_append_left (h : n < x.length) : (x ++ₛ a).get n = x[n] := by
+  induction' x with b x ih generalizing n
+  · simp at h
+  · rcases n with (_ | n)
+    · simp
+    · simp [ih n (by simpa using h), cons_append_stream]
+
+@[simp] lemma get_append_right : (x ++ₛ a).get (x.length + n) = a.get n := by
+  induction' x <;> simp [Nat.succ_add, *, cons_append_stream]
+
+@[simp] lemma get_append_length : (x ++ₛ a).get x.length = a.get 0 := get_append_right 0 x a
+
+lemma append_left_injective (h : x ++ₛ a = x ++ₛ b) : a = b := by
+  ext n; replace h := congr_arg (fun a ↦ a.get (x.length + n)) h; simpa using h
```

Comment:
```suggestion
lemma append_right_injective : Function.Injective (x ++ₛ ·) := by
  intro a b h
  ext n; replace h := congr_arg (fun a ↦ a.get (x.length + n)) h; simpa using h
```

compare to `List.append_right_injective`. (arguably, I think it should be `List.append_injective_right` but that's out-of-scope here)